### PR TITLE
ROS2DemandGraphNode Demand Changed Callback + Tests

### PR DESCRIPTION
**Pull request by TomaszTB**
Add tests and demand changed callback

### CODE REVIEW

File: ihmc-communication/src/main/java/us/ihmc/communication/ros2/ROS2DemandGraphNode.java [hunk 1]
```diff
@@ -4,20 +4,45 @@
...
+      demandChangedCallbacks.add(demandChangedCallback);
+      for (ROS2DemandGraphNode dependent : dependents)
+         dependent.addDemandChangedCallback(this::checkIfDemandChanged);
+   }
+
+   private void checkIfDemandChanged(boolean ignored)
+   {
+      boolean isDemanded = isDemanded();
+      // if wasDemanded != isDemanded -> wasDemanded = isDemanded, trigger callbacks
+      if (wasDemanded.compareAndSet(!isDemanded, isDemanded))
```
Comment (TomaszTB) [inline ihmc-communication/src/main/java/us/ihmc/communication/ros2/ROS2DemandGraphNode.java:41]: This line is confusing to me, so I added the above comment... but is the comment confusing too?

Comment (ds58) [reply ihmc-communication/src/main/java/us/ihmc/communication/ros2/ROS2DemandGraphNode.java:41]: Maybe write it out in plain english?

**Review by ds58 — APPROVED** [review]
